### PR TITLE
Alerts v2

### PR DESCRIPTION
**Pull request by emschwartz**
This PR completely changes the way we do alerts. (Note, however, that 5,500 lines of the diff is from a generated rules file!)

Instead of generating a custom alerting and recording rules file from the code, we now use a single file for all autometrics libraries.

The way this works is that the rules file defines a bunch of recording and alerting rules that are, by default, dormaint. The rules are enabled when specific labels are attached to some of the metrics produced by autometrics.

This massively simplifies supporting alerting across the various autometrics implementations because we only need to have them attach certain labels to the metrics, instead of needing to reimplement the alert generation logic in every library.

Thanks to @hatchan and @keturiosakys for their help rethinking how alerting should work and on the library API.

### CODE REVIEW

File: autometrics-macros/src/lib.rs [hunk 1]
```diff
@@ -222,24 +177,43 @@ fn instrument_function(args: &Args, item: ItemFn) -> Result<TokenStream> {
...
+                    CALLER.get(),
+                    result_labels,
+                    #objective,
+                )
             }
         }
     };
 
+    let gauge_labels = if args.track_concurrency {
+        quote! { Some(&autometrics::__private::GaugeLabels { function: #function_name, module: module_path!() }) }
```
Comment (mellowagain) [inline autometrics-macros/src/lib.rs:212]: `module_path!` 🙌

File: examples/util/src/lib.rs [hunk 2]
```diff
@@ -1,11 +1,20 @@
...
 use tokio::time::sleep;
 
 const PROMETHEUS_CONFIG_PATH: &str = concat!(env!("CARGO_MANIFEST_DIR"), "/prometheus.yml");
 
-pub fn run_prometheus() {
+pub struct ChildGuard(Child);
+
+impl Drop for ChildGuard {
+    fn drop(&mut self) {
+        self.0.kill().expect("Failed to kill prometheus");
```
Comment (mellowagain) [inline examples/util/src/lib.rs:12]: `panic!`ing on `drop` implementations is really nasty (will most likely abort) but that seems fine in this case

**Review by mellowagain — APPROVED** [review]

File: examples/util/src/lib.rs (hunk 2 above)
Comment (emschwartz) [reply examples/util/src/lib.rs:12]: Should we just swallow the error instead?

Comment (mellowagain) [reply examples/util/src/lib.rs]: we shoudl just handler the error, e.g print out the error that was returned and probably also the pid of the child process so the user can investigate it easier

File: README.md [hunk 3]
```diff
@@ -72,32 +72,12 @@ pub fn get_metrics() -> (StatusCode, String) {
...
-```rust
-fn print_prometheus_alerts() {
-  println!("{}", autometrics::generate_alerts());
-}
-```
-
-Take a look at the [alerts example](./examples/alerts) to see how to integrate generating the alert definitions into your Clap-based binary.
-
-Refer to the Prometheus docs section on [Alerting](https://prometheus.io/docs/alerting/latest/overview/) for more details on configuring Prometheus to use the alerting rules and on how to use [Alertmanager](https://prometheus.io/docs/alerting/latest/alertmanager/) to de-duplicate alerts.
+See the API docs for creating an using an autometrics [Objective](https://docs.rs/autometrics/latest/autometrics/struct.Objective.html). You can also see it the [full API example](./examples/full-api).
```
Comment (keturiosakys) [inline README.md:80]: I think a short example in the README here would help

**Pull request merged**